example_queries/get_service_name.rs.ref

```diff
@@ -181,83 +254,76 @@ impl Filter {
  
     }
 
-    pub fn create_trace_graph(&mut self, mut mod_rpc: Rpc) -> Graph<(String, IndexMap<String, String>), String> {
-        let trace;
-        let mut path = mod_rpc.headers["properties_path"].clone();
-        let mut properties = Vec::new();
-        for header in mod_rpc.headers.keys() {
-            if header.contains("properties_") && header != "properties_path" {
-                properties.push(mod_rpc.headers[header].clone().replace("properties_", ""));
-            }
-        }
-        let prop_to_trace = properties.join(",");
-        trace = utils::generate_trace_graph_from_headers(path, prop_to_trace);
-        return trace;
-    }
-
     pub fn on_incoming_requests(&mut self, mut x: Rpc) -> Vec<Rpc> {
-        let mut prop_str;
-        prop_str = format!("{whoami}.{property}=={value}",
-                                                      whoami=&self.whoami.as_ref().unwrap(),
-                                                      property="service_name",
-                                                      value=self.filter_state["node.metadata.WORKLOAD_NAME"].string_data.as_ref().unwrap().to_string());
-                                             
-        if x.headers.contains_key("properties_service_name") {
-            if !x.headers["properties_service_name"].contains(&prop_str) { // don't add our properties if they have already been added
-                x.headers.get_mut(&"properties_service_name".to_string()).unwrap().push_str(",");
-                x.headers.get_mut(&"properties_service_name".to_string()).unwrap().push_str(&prop_str);
-            }
-        }
-        else {
-            x.headers.insert("properties_service_name".to_string(), prop_str);
-        }
-         prop_str = format!("{whoami}.{property}=={value}",
-                                                      whoami=&self.whoami.as_ref().unwrap(),
-                                                      property="service_name",
-                                                      value=self.filter_state["node.metadata.WORKLOAD_NAME"].string_data.as_ref().unwrap().to_string());
-                                             
-        if x.headers.contains_key("properties_service_name") {
-            if !x.headers["properties_service_name"].contains(&prop_str) { // don't add our properties if they have already been added
-                x.headers.get_mut(&"properties_service_name".to_string()).unwrap().push_str(",");
-                x.headers.get_mut(&"properties_service_name".to_string()).unwrap().push_str(&prop_str);
+        // Fetch ferried data
+        let mut ferried_data: FerriedData;
+        if !x.headers.contains_key("ferried_data") {
```

Comment:
This will not work on the filter for example.

Reply:
Wouldn't it be equivalent to 

if let Some(ferried_data) = self.get_http_request_header("ferried_data") {
    // do some stuff
} else  {
    // do other stuff
}

That's what we use to get the trace ID, so I thought it would be the same.